documentation/specs/src/interoperability/ethereum-bridge/transfers_to_ethereum.md

```diff
@@ -123,11 +125,46 @@ If vote extensions are not available, this signed root may lag behind the
 list of pending transactions. However, it should be the eventually every 
 pending transaction is covered by the root or it times out.
 
-## Replay Protection and timeouts
-
-It is important that nonces are used to prevent copies of the same
-transaction being submitted multiple times. Since we do not want to enforce
-an order on the transactions, these nonces should land in a range. As a
-consequence of this, it is possible that transactions in the Bridge Pool will
-time out. Transactions that timed out should revert the state changes on
-Namada including refunding the paid in fees.
+## Replay Protection
+
+State updates in Namada, derived from transfer to Ethereum events, result in
+the increment of the value of a monotonically growing nonce, that is signed
+together with the root of the Bridge Pool. The `Bridge` smart contract's nonce
+of transfers to Ethereum is kept in sync with Namada's, upon relay calls. This
+behavior should contribute to avoiding replay attacks on Ethereum events related
+to transfers to Ethereum, as it assigns a unique id on a snapshot of the Bridge
+Pool. Ethereum events arriving at the ledger with a nonce different from the next
+expected nonce are rejected.
+
+## Timeouts
+
+Pending transfers to Ethereum sitting in the Bridge Pool for some duration of
+$T_{dur}$ blocks are subject to timeouts. The exact value of $T_{dur}$, for
+now, is the minimum duration in blocks of an epoch. Transactions that time out
+should revert the state changes in Namada, including refunding the paid fees
+and escrowed assets.
+
+Timeouts only happen after we process transfer to Ethereum events to guarantee
```

Comment:
What if a particular bridge pool update is never relayed to Ethereum, though? Can we ever claim the assets back on Namada?

Reply:
I guess if we require ordered processing of bridge pool updates anyways, there's no way to recover from this anyways. This does mean that we'll be less able to be responsive to Ethereum gas markets, though :thinking:. Just curious, did we consider a design where the timeouts reference Ethereum-side block heights, and processing a particular transfer after that block height will fail + you can make a proof that a transfer was not processed before that block height (emitting an Ethereum event which can be confirmed on Namada, releasing the assets)?

Reply:
you can pick a subset of whichever transfers from the bridge pool you want to relay. quorum certificates contain signatures over the merkle root of the bridge pool and some nonce. you provide an inclusion proof for the subset you want to relay. so, there should always be relays happening, regardless of whether a particular transfer is included or not. eventually, transfers that are never relayed will time out, and the assets and fees can be reclaimed back on Namada.

we did not consider timing out events based on their Ethereum-side block height, to answer your question

Reply:
I'm less concerned with Namada not reaching a quorum decision on a particular nonce than I am with the Ethereum gas prices getting high and no one wanting to relay batches, during which a large queue builds up because the Ethereum side won't allow later batches tto be relayed until earlier ones have been - I'm assuming that the Ethereum contracts check the nonce and increment it each time, right?

Reply:
> I'm assuming that the Ethereum contracts check the nonce and increment it each time, right?

yup

this is definitely an issue that affects throughput. the escrowed funds would not be able to be reclaimed, with the current design, since they either need to time out or be relayed to be released from the bridge pool, and this only happens when we get a transfer to Ethereum event

Reply:
alright - I think this is OK for now, but let's definitely note it down